app/controllers/tasks.js

```diff
@@ -262,16 +276,70 @@ export default class TasksController extends Controller {
       );
     }
   }
-
-  @action async handleUpdateTask(taskId, error) {
+  showTaskChangeInfoModal(msg) {
+    this.message = msg;
+    this.showModal = true;
+    this.buttonRequired = true;
+  }
+  @action async handleUpdateTask(taskId, resetCurrentTask) {
+    this.resetCurrentTask = resetCurrentTask;
     const taskData = this.taskFields;
-    if (taskData.percentCompleted === TASK_PERCENTAGE.completedPercentage) {
-      this.message = TASK_MESSAGES.MARK_DONE;
-      this.showModal = true;
-      this.buttonRequired = true;
-      this.tempTaskId = taskId;
+    const currentTask = this.getTaskById(taskId);
+    const isCurrentTaskStatusBlock =
+      currentTask.status === this.TASK_KEYS.BLOCKED;
+    const isNewTaskStatusInProgress =
+      taskData.status === this.TASK_KEYS.IN_PROGRESS;
+    const isNewTaskStatusBlock = taskData.status === this.TASK_KEYS.BLOCKED;
+    const isCurrProgress100 =
+      parseInt(currentTask.percentCompleted || 0) === 100;
+    const isCurrProgress0 = parseInt(currentTask.percentCompleted || 0) === 0;
+    this.tempTaskId = taskId;
+
+    if (this.dev && taskData.status) {
+      if (!isCurrProgress100) {
+        switch (currentTask.status) {
+          case this.TASK_KEYS.IN_PROGRESS:
+            if (!isNewTaskStatusBlock) {
+              this.showTaskChangeInfoModal(
+                TASK_MESSAGES.CHANGE_TO_100_PROGRESS
+              );
+              this.taskFields.percentCompleted = 100;
+              return;
+            }
+            break;
```

Comment:
my suggestion was to move those variables out of the if statements, not to create a separate function but it looks good
you can move this.taskFields.percentCompleted = 100; out of the switch case and remove unnecessary break statements as the case will end as soon as you return